src/tests/profiler/gc/nongcheap.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Profiler.Tests
+{
+    class NonGCHeapTests
+    {
+        static readonly Guid GcAllocateEventsProfilerGuid = new Guid("EF0D191C-3FC7-4311-88AF-E474CBEB2859");
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void AllocateNonGcHeapObjects()
+        {
+            // When this method is invoked, JIT is expected to trigger allocations for these
```

Comment:
This might not work for ReadyToRun?

Reply:
Thanks, disabled the test for crossgen

Reply:
I do not think that this is correct. Did you see the test actually failing with crossgen?

If I am reading the code correctly, this test should work with crossgen just fine. The string literals referenced by R2R code must be allocated on frozen heap. Otherwise, the tiering from R2R to Tier1 would work poorly - the tiering would be stuck with non-frozen string literals.

Reply:
Thanks, you're absolutely right